.env.d.ts

```diff
@@ -1 +0,0 @@
-/// <reference types="@histoire/plugin-vue/components" />
```

Comment:
- moved this to `env.d.ts`, along with content from `lib/types/shims.d.ts`
- needed to explicitly specify .d.ts in path there
- needed to explicitly install `@vue/runtime-core` for module augmentation to work (weirdness because of pnpm, other option was to set `shamefully-hoist=true` in `.npmrc`)
- needed to explicitly add it to `tsconfig.includes`
- removed `declare module '*.vue'` as it seem to be no longer needed and it was suppressing TS errors for missing components, wrong paths, typos, difference in case, etc.
- moving shims outside lib shouldn't have any effect on other projects because they already have vite/client referenced in their respective shims

Volar is now picking up Histoire components like HstSelect, HstText, HstNumber, Story, ... 🎉